## 49% of end-end tests have passed
```
Total executed tests: 51
Failed tests: 24
Expected to fail tests: 2
Passed tests: 25
```
```
IDs of failed tests: 727230,703086,702786,703629,702948,702838,702859,702809,702843,703194,702844,702839,702733,702894,702958,702957,702841,702845,702775,704613,704615,727229,702947,702840 
```
```
IDs of expected to fail tests: 703495,727232 
```
<h3>Failed tests (24)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=seanstrom/process-keypair-data-signal&PR_ID=20550&APK_NAME=20550.apk&TR_CASE_IDS=727230,703086,702786,703629,702948,702838,702859,702809,702843,703194,702844,702839,702733,702894,702958,702957,702841,702845,702775,704613,704615,727229,702947,702840">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592330">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div></blockquote></p>

```
critical/test_wallet.py:154: in test_wallet_send_asset_from_drawer
    self.wallet_2.chats_tab.click()
../views/base_element.py:90: in click
    element = self.find_element()
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 2: ChatsTab by accessibility id: `chats-stack-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592329">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `top-bar` </div><div>Device 2: Find `Button` by `accessibility id`: `top-bar` </div></blockquote></p>

```
critical/test_wallet.py:147: in test_wallet_send_eth
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.0023260521 but expected to be 0.0023281521
E    Can't find the last transaction for the receiver
```

<br/><br/></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592288">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `BrowserTab` by `accessibility id`: `browser-stack-tab` </div><div>Device 1: Tap on found: BrowserTab </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:115: in test_links_deep_links
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Open community for e2e' was not requested to join by the deep link status.app://c/GzwAAGS9TbI9CiYaoZj0TRNelWAPwk8gSHIEmqgqb0ADPw5rnFvk6bp_n3BlZxVMshmxFyI=#zQ3shkrmSmwb66sw2EuUvC5iqiNqEeevTfjtks3p8ZaaCYoNj
E    Community 'Closed community' was not requested to join by the deep link status.app://c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4215df185552415c89d8a201e929b3f9?auth=f0a1b1ca67fa9cda16ee23f60e6ae762#201">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592327">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><p><blockquote><div>Device 1: Looking for a message by text: https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK </div><div>Device 1: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK')]` </div></blockquote></p>

```
critical/test_deep_and_universal_links.py:71: in test_links_open_universal_links_from_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Community 'Open community for e2e' was not requested to join by the url https://status.app/c/GzwAAGS9TbI9CiYaoZj0TRNelWAPwk8gSHIEmqgqb0ADPw5rnFvk6bp_n3BlZxVMshmxFyI=#zQ3shkrmSmwb66sw2EuUvC5iqiNqEeevTfjtks3p8ZaaCYoNj
E    Community 'Closed community' was not requested to join by the url https://status.app/c/GzAAAORtwyW4xNWM4td0F7hOnYZ1apSqCCRUUR0qxD19n3Ec97fX_aIVIGFWbdUM#zQ3shk6dgK8dYWWSC4m8Jj5c91zyfhfj1fFkgypS8D9gsXkrK
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4215df185552415c89d8a201e929b3f9?auth=f0a1b1ca67fa9cda16ee23f60e6ae762#52">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592315">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td>

```
Test setup failed: activity_center/test_activity_center.py:414: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592314">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` </div><div>Device 2: Wait for element `Button` for max 120s and click when it is available </div></blockquote></p>

```
Test setup failed: activity_center/test_activity_center.py:414: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b052148097ed4ac793446679e325cb11?auth=961c10e2784863ee9d1269e93f7be29a">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8ffab2bbe20f4c51afd9f87924728f58?auth=e23afe29250d166eb14a242249054647">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592316">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592298">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` </div><div>Device 2: Wait for element `Button` for max 120s and click when it is available </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/246d13544f56417c8be47d32247ade69?auth=116cd0528b459aad7a8e70f2a3e1738b">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ee5ae8a390e44799c26f75f202e5afe?auth=d41161f26448adc9ab612038e99176a5">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592309">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592302">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592318">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592303">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592299">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592311">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592301">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592328">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592300">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:347: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592312">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` </div><div>Device 2: Wait for element `Button` for max 120s and click when it is available </div></blockquote></p>

```
Test setup failed: activity_center/test_activity_center.py:274: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3bb617a8b25249d6a04767f91b81be93?auth=c6f080b90568270b0a24c3b60b9196b8">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2a691c45283e453bba7c8d12ee7ecf70?auth=7d1904aba37300d51c41e638499fd0c1">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592293">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='chat-item']//android.widget.TextView[contains(@text,'https://status.app/c/')]` </div><div>Device 2: Wait for element `Button` for max 120s and click when it is available </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0912892a6d3d4461a84c7986f3067f86?auth=57949bc694d067fd8d84d85ce7738267">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cbe6863b4e924b68adec91220186c607?auth=c92c3d9c923c4f79c18cce10b4e6752b">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592326">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[Can't enter channel after community is fetched for the first time, https://github.com/status-im/status-mobile/issues/20395]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592313">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element 
```

<b>[[Can't enter channel after community is fetched for the first time, https://github.com/status-im/status-mobile/issues/20395]]</b><br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592296">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592304">test_community_leave</a>, id: 702845 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:867: in prepare_devices
    self.community_2.join_community()
../views/chat_view.py:420: in join_community
    self.join_button.wait_and_click(120)
../views/base_element.py:100: in wait_and_click
    self.wait_for_visibility_of_element(sec)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by accessibility id:`show-request-to-join-screen-button` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592285">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td>

```
critical/chats/test_1_1_public_chats.py:469: in test_1_1_chat_text_message_delete_push_disappear
    if not self.chat_2.chat_message_input.is_element_displayed():
../views/base_element.py:221: in is_element_displayed
    return self.wait_for_visibility_of_element(sec, ignored_exceptions=ignored_exceptions)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.helpers.doFindElementOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/find.js:30:10)
E       at doFind (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:36:17)
E       at wrappedCondFn (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:60:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#550">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#400">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (2)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592332">test_wallet_add_remove_watch_only_account</a>, id: 727232 </b></td></tr><tr><td><p><blockquote><div>Device 1: `Text` is `0x8d2413447ff297d30bdc475f6d5cb00254685aae` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/test_wallet.py:256: in test_wallet_add_remove_watch_only_account
    self.home_view.driver.fail(
base_test_case.py:178: in fail
    pytest.fail('Device %s: %s' % (self.number, text))
 Device 1: Incorrect address '0x8d2413447ff297d30bdc475f6d5cb00254685aae' is shown when swiping between accounts, expected one is '0x8d2413447ff297d30bdc475f6d5cb00254685aae' 
```

<b>[[Missing networks in account address, https://github.com/status-im/status-mobile/issues/20166]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c6126953aad94e69bd27dfccb1bc5bfe?auth=98fac670fb76ceb38d53474fcd12fafd#102">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592323">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td><p><blockquote><div># STEP: Change device time so chat will be unmuted by timer</div><div>Device 2: Long press on `ChatElement` </div></blockquote></p>

```
critical/chats/test_group_chat.py:497: in test_group_chat_mute_chat
    self.errors.verify_no_errors()
base_test_case.py:190: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Chat is still muted after timeout 
```

<b>[[Chat is not unmuted after expected time: https://github.com/status-im/status-mobile/issues/19627]]</b><br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#475">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#273">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#242">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (25)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592294">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#156">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#78">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592284">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#355">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#230">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#228">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592319">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#173">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#106">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#86">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592295">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#322">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#203">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#201">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592320">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/908787db0caa427594682d3acf0719e6?auth=d4939a2a575048af23fe1e1e4bb68e03#282">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0892ca119a5e4cf199f44f438647371d?auth=92b7da0a9202539d6ae3ecb9a0cb9875#168">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b4e3dc56dc9b4880b0a9dc8721de5cf6?auth=6e00a74228f645269411169175c0d896#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592286">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592321">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592310">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#105">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592305">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#52">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592325">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#354">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592317">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b324a372dacb411682a57ec7e0498a5f?auth=a79f1dd4bffe0fb3360ea39efad2b196#229">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592324">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8b70e3fe9b72439ca55b1b1fee5451cd?auth=15c50718eddc408ce21a3288fac2f6e1#105">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ee4d647ecf034d56b900e3ca375899a1?auth=1eb0959a176633df3b6124c1f3865354#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592291">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8b70e3fe9b72439ca55b1b1fee5451cd?auth=15c50718eddc408ce21a3288fac2f6e1#95">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ee4d647ecf034d56b900e3ca375899a1?auth=1eb0959a176633df3b6124c1f3865354#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592292">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8b70e3fe9b72439ca55b1b1fee5451cd?auth=15c50718eddc408ce21a3288fac2f6e1#169">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ee4d647ecf034d56b900e3ca375899a1?auth=1eb0959a176633df3b6124c1f3865354#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592331">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c6126953aad94e69bd27dfccb1bc5bfe?auth=98fac670fb76ceb38d53474fcd12fafd#38">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592306">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dba465fe3cd3404a8eebcba03b90e8ba?auth=dfb90a030ed7d360de6006207b31f5c2#56">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f45829be885a4887a5607202a1bf2787?auth=daccfb42ae4290a370d11960dbcaf21b#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592289">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dba465fe3cd3404a8eebcba03b90e8ba?auth=dfb90a030ed7d360de6006207b31f5c2#134">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f45829be885a4887a5607202a1bf2787?auth=daccfb42ae4290a370d11960dbcaf21b#214">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592307">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dba465fe3cd3404a8eebcba03b90e8ba?auth=dfb90a030ed7d360de6006207b31f5c2#99">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f45829be885a4887a5607202a1bf2787?auth=daccfb42ae4290a370d11960dbcaf21b#121">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592290">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#147">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#119">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592287">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#351">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#209">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592297">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#409">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#259">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592282">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#95">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#84">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592308">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#458">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#290">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592283">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#209">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#164">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3592322">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/38a132cbda584b39accc62f3b938df95?auth=b85e54e8bc1708ce22263dc9fa03d021#489">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f5817122cb104c88b7b6177bb499da4c?auth=34c55b906708eb681cb69622b3b0ddfe#340">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/chat/messenger/messages/transport/events.cljs
+++ b/src/status_im/contexts/chat/messenger/messages/transport/events.cljs
@@ -183,8 +183,8 @@
       (seq keypairs)
       (do
         (js-delete response-js "keypairs")
-        (rf/dispatch [:wallet/process-keypairs (types/js->clj keypairs)])
         (rf/merge cofx
+                  {:fx [[:dispatch [:wallet/process-keypairs (types/js->clj keypairs)]]]}
                   (process-next response-js sync-handler)))
 
       (seq settings)

--- a/src/status_im/contexts/chat/messenger/messages/transport/events.cljs
+++ b/src/status_im/contexts/chat/messenger/messages/transport/events.cljs
@@ -55,6 +55,7 @@
         ^js cleared-histories          (.-clearedHistories response-js)
         ^js identity-images            (.-identityImages response-js)
         ^js accounts                   (.-accounts response-js)
+        ^js keypairs                   (.-keypairs response-js)
         ^js ens-username-details-js    (.-ensUsernameDetails response-js)
         ^js customization-color-js     (.-customizationColor response-js)
         ^js saved-addresses-js         (.-savedAddresses response-js)
@@ -176,6 +177,13 @@
       (seq accounts)
       (do
         (js-delete response-js "accounts")
+        (rf/merge cofx
+                  (process-next response-js sync-handler)))
+
+      (seq keypairs)
+      (do
+        (js-delete response-js "keypairs")
+        (rf/dispatch [:wallet/process-keypairs (types/js->clj keypairs)])
         (rf/merge cofx
                   (process-next response-js sync-handler)))
 

--- a/src/status_im/contexts/wallet/data_store.cljs
+++ b/src/status_im/contexts/wallet/data_store.cljs
@@ -5,6 +5,7 @@
     [clojure.string :as string]
     [status-im.constants :as constants]
     [status-im.contexts.wallet.common.utils.networks :as network-utils]
+    [utils.collection :as utils.collection]
     [utils.money :as money]
     [utils.number :as utils.number]
     [utils.transforms :as transforms]))
@@ -160,3 +161,26 @@
 (defn rpc->saved-addresses
   [saved-addresses]
   (map rpc->saved-address saved-addresses))
+
+(defn process-keypairs
+  [keypairs]
+  (let [received-keypairs             (rpc->keypairs keypairs)
+        keypair-label                 #(if % :removed-keypairs :updated-keypairs)
+        {:keys [removed-keypairs
+                updated-keypairs]
+         :or   {updated-keypairs []
+                removed-keypairs []}} (group-by (comp keypair-label :removed) received-keypairs)
+        updated-keypairs-by-id        (utils.collection/index-by :key-uid updated-keypairs)
+        updated-accounts-by-address   (->> updated-keypairs
+                                           (mapcat :accounts)
+                                           (filter (comp not :chat))
+                                           (utils.collection/index-by :address))
+        removed-keypairs-ids          (set (map :key-uid removed-keypairs))
+        removed-account-addresses     (->> removed-keypairs
+                                           (mapcat :accounts)
+                                           (map :address)
+                                           (set))]
+    {:removed-keypair-ids         removed-keypairs-ids
+     :removed-account-addresses   removed-account-addresses
+     :updated-keypairs-by-id      updated-keypairs-by-id
+     :updated-accounts-by-address updated-accounts-by-address}))

--- a/src/status_im/contexts/wallet/data_store_test.cljs
+++ b/src/status_im/contexts/wallet/data_store_test.cljs
@@ -1,0 +1,193 @@
+(ns status-im.contexts.wallet.data-store-test
+  (:require
+    [cljs.test :refer-macros [deftest is testing]]
+    [matcher-combinators.matchers :as matchers]
+    matcher-combinators.test
+    [status-im.contexts.wallet.data-store :as sut]
+    [utils.collection :as utils.collection]))
+
+(def raw-account
+  {:path                  "m/44'/60'/0'/0/0"
+   :name                  "Account name"
+   :wallet                true
+   :chat                  false
+   :emoji                 "🍙"
+   :colorId               "blue"
+   :type                  "generated"
+   :createdAt             1716548742000
+   :prodPreferredChainIds "1:42161"
+   :testPreferredChainIds "11155111:421614"
+   :removed               false
+   :operable              "fully"})
+
+(defn make-raw-account
+  [overrides]
+  (merge raw-account overrides))
+
+(def raw-keypair-profile
+  {:type     "profile"
+   :key-uid  "0x000"
+   :accounts [(make-raw-account {:key-uid "0x000"
+                                 :address "1x000"
+                                 :wallet  false
+                                 :chat    true})
+              (make-raw-account {:key-uid "0x000"
+                                 :address "2x000"})]})
+
+(def raw-keypair-seed-phrase
+  {:type     "seed"
+   :key-uid  "0x123"
+   :accounts [(make-raw-account {:key-uid "0x123"
+                                 :address "1x123"})]})
+
+(def raw-keypair-private-key
+  {:type     "key"
+   :key-uid  "0x456"
+   :accounts [(make-raw-account {:key-uid  "0x456"
+                                 :address  "1x456"
+                                 :operable "no"})]})
+
+(deftest chain-ids-string->set-test
+  (testing "chaind-ids-string->set splits and parses chain-ids from string"
+    (is (match? #{1 42161}
+                (sut/chain-ids-string->set (:prodPreferredChainIds raw-account))))
+    (is (match? #{11155111 421614}
+                (sut/chain-ids-string->set (:testPreferredChainIds raw-account))))))
+
+(deftest rpc->keypair-test
+  (testing "rpc->keypair transforms a profile keypair"
+    (is
+     (match? {:type               :profile
+              :key-uid            "0x000"
+              :lowest-operability :fully
+              :accounts           [{:key-uid                  "0x000"
+                                    :operable                 :fully
+                                    :chat                     true
+                                    :wallet                   false
+                                    :address                  "1x000"
+                                    :path                     (:path raw-account)
+                                    :name                     (:name raw-account)
+                                    :removed                  (:removed raw-account)
+                                    :color                    (keyword (:colorId raw-account))
+                                    :created-at               (:createdAt raw-account)
+                                    :prod-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:prodPreferredChainIds raw-account))
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds raw-account))}
+                                   {:key-uid                  "0x000"
+                                    :operable                 :fully
+                                    :chat                     false
+                                    :wallet                   true
+                                    :address                  "2x000"
+                                    :path                     (:path raw-account)
+                                    :name                     (:name raw-account)
+                                    :removed                  (:removed raw-account)
+                                    :color                    (keyword (:colorId raw-account))
+                                    :created-at               (:createdAt raw-account)
+                                    :prod-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:prodPreferredChainIds raw-account))
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds raw-account))}]}
+             (sut/rpc->keypair raw-keypair-profile))))
+  (testing "rpc->keypair transforms a seed-phrase keypair"
+    (is
+     (match? {:type               :seed
+              :key-uid            "0x123"
+              :lowest-operability :fully
+              :accounts           [{:key-uid                  "0x123"
+                                    :address                  "1x123"
+                                    :operable                 :fully
+                                    :path                     (:path raw-account)
+                                    :name                     (:name raw-account)
+                                    :wallet                   (:wallet raw-account)
+                                    :removed                  (:removed raw-account)
+                                    :color                    (keyword (:colorId raw-account))
+                                    :created-at               (:createdAt raw-account)
+                                    :prod-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:prodPreferredChainIds raw-account))
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds raw-account))}]}
+             (sut/rpc->keypair raw-keypair-seed-phrase))))
+  (testing "rpc->keypair transforms a raw private-key keypair with inoperable accounts"
+    (is
+     (match? {:type               :key
+              :key-uid            "0x456"
+              :lowest-operability :no
+              :accounts           [{:key-uid                  "0x456"
+                                    :address                  "1x456"
+                                    :operable                 :no
+                                    :path                     (:path raw-account)
+                                    :name                     (:name raw-account)
+                                    :wallet                   (:wallet raw-account)
+                                    :removed                  (:removed raw-account)
+                                    :color                    (keyword (:colorId raw-account))
+                                    :created-at               (:createdAt raw-account)
+                                    :prod-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:prodPreferredChainIds raw-account))
+                                    :test-preferred-chain-ids (sut/chain-ids-string->set
+                                                               (:testPreferredChainIds
+                                                                raw-account))}]}
+             (sut/rpc->keypair raw-keypair-private-key)))))
+
+(deftest rpc->keypairs-test
+  (testing "rpc->keypairs transforms and sorts raw keypairs"
+    (is
+     (match? [(sut/rpc->keypair raw-keypair-profile)
+              (sut/rpc->keypair raw-keypair-seed-phrase)
+              (sut/rpc->keypair raw-keypair-private-key)]
+             (sut/rpc->keypairs [raw-keypair-seed-phrase
+                                 raw-keypair-private-key
+                                 raw-keypair-profile])))))
+
+(deftest process-keypairs-test
+  (testing "process-keypairs represents updated key pairs and accounts"
+    (is
+     (match?
+      (matchers/match-with
+       [set? matchers/set-equals
+        map? matchers/equals]
+       {:removed-keypair-ids         #{}
+        :removed-account-addresses   #{}
+        :updated-accounts-by-address (->>
+                                       [raw-keypair-seed-phrase
+                                        raw-keypair-private-key]
+                                       (sut/rpc->keypairs)
+                                       (mapcat :accounts)
+                                       (utils.collection/index-by :address))
+        :updated-keypairs-by-id      {(:key-uid raw-keypair-seed-phrase)
+                                      (sut/rpc->keypair raw-keypair-seed-phrase)
+                                      (:key-uid raw-keypair-private-key)
+                                      (sut/rpc->keypair raw-keypair-private-key)}})
+      (sut/process-keypairs [raw-keypair-seed-phrase
+                             raw-keypair-private-key]))))
+  (testing "process-keypairs represents removed key pairs and accounts"
+    (is (match?
+         (matchers/match-with
+          [set? matchers/set-equals
+           map? matchers/equals]
+          {:removed-keypair-ids         #{"0x456"}
+           :removed-account-addresses   #{"1x456"}
+           :updated-accounts-by-address (->> [raw-keypair-seed-phrase]
+                                             (sut/rpc->keypairs)
+                                             (mapcat :accounts)
+                                             (utils.collection/index-by :address))
+           :updated-keypairs-by-id      {(:key-uid raw-keypair-seed-phrase)
+                                         (sut/rpc->keypair raw-keypair-seed-phrase)}})
+         (sut/process-keypairs [raw-keypair-seed-phrase
+                                (assoc raw-keypair-private-key :removed true)]))))
+  (testing "process-keypairs ignores chat accounts inside updated accounts"
+    (is
+     (match?
+      (matchers/match-with
+       [set? matchers/set-equals
+        map? matchers/equals]
+       {:removed-keypair-ids         #{}
+        :removed-account-addresses   #{}
+        :updated-accounts-by-address (->> [raw-keypair-profile]
+                                          (sut/rpc->keypairs)
+                                          (mapcat :accounts)
+                                          (filter (comp not :chat))
+                                          (utils.collection/index-by :address))
+        :updated-keypairs-by-id      {(:key-uid raw-keypair-profile)
+                                      (sut/rpc->keypair raw-keypair-profile)}})
+      (sut/process-keypairs [raw-keypair-profile])))))

--- a/src/status_im/contexts/wallet/events.cljs
+++ b/src/status_im/contexts/wallet/events.cljs
@@ -1,6 +1,7 @@
 (ns status-im.contexts.wallet.events
   (:require
     [camel-snake-kebab.extras :as cske]
+    [clojure.set]
     [clojure.string :as string]
     [react-native.platform :as platform]
     [status-im.constants :as constants]
@@ -81,6 +82,11 @@
 
 (rf/reg-event-fx :wallet/log-rpc-error log-rpc-error)
 
+(def refresh-accounts-fx-dispatches
+  [[:dispatch [:wallet/get-wallet-token-for-all-accounts]]
+   [:dispatch [:wallet/request-collectibles-for-all-accounts {:new-request? true}]]
+   [:dispatch [:wallet/check-recent-history-for-all-accounts]]])
+
 (rf/reg-event-fx
  :wallet/get-accounts-success
  (fn [{:keys [db]} [accounts]]
@@ -91,11 +97,9 @@
      {:db (assoc-in db
            [:wallet :accounts]
            (utils.collection/index-by :address wallet-accounts))
-      :fx [[:dispatch [:wallet/get-wallet-token-for-all-accounts]]
-           [:dispatch [:wallet/request-collectibles-for-all-accounts {:new-request? true}]]
-           [:dispatch [:wallet/check-recent-history-for-all-accounts]]
-           (when new-account?
-             [:dispatch [:wallet/navigate-to-new-account navigate-to-account]])]})))
+      :fx (concat refresh-accounts-fx-dispatches
+                  [(when new-account?
+                     [:dispatch [:wallet/navigate-to-new-account navigate-to-account]])])})))
 
 (rf/reg-event-fx
  :wallet/get-accounts
@@ -545,3 +549,36 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn process-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypair-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/process-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (nil? (get updated-accounts-by-address
+                                                                             address)))))
+                                                   (map :address))]
+    (cond-> {:db (-> db
+                     (assoc-in [:wallet :keypairs]
+                               (-> (partial dissoc existing-keypairs-by-id)
+                                   (apply removed-keypair-ids)
+                                   (merge updated-keypairs-by-id)))
+                     (assoc-in [:wallet :accounts]
+                               (-> (partial dissoc existing-accounts-by-address)
+                                   (apply (into removed-account-addresses old-account-addresses))
+                                   (merge updated-accounts-by-address))))}
+      (seq new-account-addresses)
+      (assoc :fx refresh-accounts-fx-dispatches))))
+
+(rf/reg-event-fx :wallet/process-keypairs process-keypairs)

--- a/src/status_im/contexts/wallet/events.cljs
+++ b/src/status_im/contexts/wallet/events.cljs
@@ -572,12 +572,12 @@
                                                    (map :address))]
     (cond-> {:db (-> db
                      (assoc-in [:wallet :keypairs]
-                               (-> (partial dissoc existing-keypairs-by-id)
-                                   (apply removed-keypair-ids)
+                               (-> (apply dissoc existing-keypairs-by-id removed-keypair-ids)
                                    (merge updated-keypairs-by-id)))
                      (assoc-in [:wallet :accounts]
-                               (-> (partial dissoc existing-accounts-by-address)
-                                   (apply (into removed-account-addresses old-account-addresses))
+                               (-> (apply dissoc
+                                          existing-accounts-by-address
+                                          (into removed-account-addresses old-account-addresses))
                                    (merge updated-accounts-by-address))))}
       (seq new-account-addresses)
       (assoc :fx refresh-accounts-fx-dispatches))))

--- a/src/status_im/contexts/wallet/events_test.cljs
+++ b/src/status_im/contexts/wallet/events_test.cljs
@@ -1,6 +1,7 @@
 (ns status-im.contexts.wallet.events-test
   (:require
     [cljs.test :refer-macros [is testing]]
+    [matcher-combinators.matchers :as matchers]
     matcher-combinators.test
     [re-frame.db :as rf-db]
     [status-im.constants :as constants]
@@ -150,3 +151,144 @@
                                [:dispatch [:wallet/check-recent-history-for-account address]]]}]
     (reset! rf-db/app-db {:wallet {:accounts {}}})
     (is (match? expected-effects (dispatch [event-id raw-account])))))
+
+(h/deftest-event :wallet/process-keypairs
+  [event-id dispatch]
+  (let [keypair-key-uid (:key-uid raw-account)]
+    (testing "event adds new key pairs"
+      (reset! rf-db/app-db {:wallet {:accounts {}
+                                     :keypairs {}}})
+      (is
+       (match?
+        (matchers/match-with
+         [set? matchers/set-equals
+          vector? matchers/equals
+          map? matchers/equals]
+         {:db {:wallet {:accounts {(:address account) account}
+                        :keypairs {keypair-key-uid {:key-uid            keypair-key-uid
+                                                    :type               :seed
+                                                    :lowest-operability :fully
+                                                    :accounts           [account]}}}}
+          :fx [[:dispatch [:wallet/get-wallet-token-for-all-accounts]]
+               [:dispatch [:wallet/request-collectibles-for-all-accounts {:new-request? true}]]
+               [:dispatch [:wallet/check-recent-history-for-all-accounts]]]})
+        (dispatch [event-id
+                   [{:key-uid  keypair-key-uid
+                     :type     "seed"
+                     :accounts [raw-account]}]]))))
+    (testing "event removes key pairs and accounts that are marked as removed"
+      (reset! rf-db/app-db {:wallet {:accounts {(:address account) account}
+                                     :keypairs {keypair-key-uid
+                                                {:key-uid            keypair-key-uid
+                                                 :type               :seed
+                                                 :lowest-operability :fully
+                                                 :accounts           [account]}}}})
+
+      (is
+       (match?
+        (matchers/match-with
+         [set? matchers/set-equals
+          vector? matchers/equals
+          map? matchers/equals]
+         {:db {:wallet {:accounts {}
+                        :keypairs {}}}})
+        (dispatch [event-id
+                   [{:key-uid  keypair-key-uid
+                     :type     "seed"
+                     :removed  true
+                     :accounts [raw-account]}]]))))
+    (testing "event removes accounts not present with key pair"
+      (reset! rf-db/app-db {:wallet {:accounts {(:address account) account
+                                                "1x001"            (assoc account
+                                                                          :address "1x001"
+                                                                          :key-uid "0x001")}
+                                     :keypairs {keypair-key-uid
+                                                {:key-uid            keypair-key-uid
+                                                 :type               :seed
+                                                 :lowest-operability :fully
+                                                 :accounts           [account]}
+                                                "0x001"
+                                                {:key-uid            "0x001"
+                                                 :type               :seed
+                                                 :lowest-operability :fully
+                                                 :accounts           [(assoc account
+                                                                             :address "1x001"
+                                                                             :key-uid "0x001")]}}}})
+      (is
+       (match?
+        (matchers/match-with
+         [set? matchers/set-equals
+          vector? matchers/equals
+          map? matchers/equals]
+         {:db {:wallet {:accounts {(:address account) account}
+                        :keypairs {keypair-key-uid
+                                   {:key-uid            keypair-key-uid
+                                    :type               :seed
+                                    :lowest-operability :fully
+                                    :accounts           [account]}
+                                   "0x001"
+                                   {:key-uid            "0x001"
+                                    :type               :seed
+                                    :lowest-operability :fully
+                                    :accounts           []}}}}})
+        (dispatch [event-id
+                   [{:key-uid  "0x001"
+                     :type     "seed"
+                     :accounts []}]]))))
+    (testing "event updates existing key pairs"
+      (reset! rf-db/app-db {:wallet
+                            {:accounts {(:address account)
+                                        (assoc account :operable :no)}
+                             :keypairs {keypair-key-uid
+                                        {:key-uid            keypair-key-uid
+                                         :type               :seed
+                                         :lowest-operability :no
+                                         :accounts           [(assoc account :operable :no)]}}}})
+      (is
+       (match?
+        (matchers/match-with
+         [set? matchers/set-equals
+          vector? matchers/equals
+          map? matchers/equals]
+         {:db {:wallet {:accounts {(:address account) account}
+                        :keypairs {keypair-key-uid
+                                   {:key-uid            keypair-key-uid
+                                    :type               :seed
+                                    :lowest-operability :fully
+                                    :accounts           [account]}}}}})
+        (dispatch [event-id
+                   [{:key-uid  keypair-key-uid
+                     :type     "seed"
+                     :accounts [raw-account]}]]))))
+    (testing "event ignores chat accounts for key pairs"
+      (reset! rf-db/app-db {:wallet {:accounts {(:address account) account}
+                                     :keypairs {keypair-key-uid
+                                                {:key-uid            keypair-key-uid
+                                                 :type               :profile
+                                                 :lowest-operability :fully
+                                                 :accounts           [account
+                                                                      (assoc account
+                                                                             :address "1x001"
+                                                                             :chat    true)]}}}})
+      (is
+       (match?
+        (matchers/match-with
+         [set? matchers/set-equals
+          vector? matchers/equals
+          map? matchers/equals]
+         {:db {:wallet {:accounts {(:address account) account}
+                        :keypairs {keypair-key-uid
+                                   {:key-uid            keypair-key-uid
+                                    :type               :profile
+                                    :lowest-operability :fully
+                                    :accounts           [account
+                                                         (assoc account
+                                                                :address "1x001"
+                                                                :chat    true)]}}}}})
+        (dispatch [event-id
+                   [{:key-uid  keypair-key-uid
+                     :type     "profile"
+                     :accounts [raw-account
+                                (assoc raw-account
+                                       :address "1x001"
+                                       :chat    true)]}]]))))))

